leetcodePython/Stack/stack_84.py

```diff
@@ -0,0 +1,21 @@
+from typing import List
+
+
+class Solution:
+    def largestRectangleArea(self, heights: List[int]) -> int:
+        increase_stack = [-1]
+        max_area = 0
+        for i in range(len(heights)):
+            while increase_stack[-1] != -1 and heights[increase_stack[-1]] >= heights[i]:
+                current_height = heights[increase_stack.pop()]
+                prev_height_index = increase_stack[-1]
+                current_width = i - prev_height_index - 1
```

Comment:
in the stack, it is the next smaller height index, it means:
* between `current index` and the `smaller hight index`, every number is bigger than the `current height`.
* Now that forms a `local max` area